File: MakeLifeBetter/shared/src/commonMain/kotlin/com/carlosnicolaugalves/makelifebetter/event/EventResult.kt
package com.carlosnicolaugalves.makelifebetter.event

import com.carlosnicolaugalves.makelifebetter.model.Event
import com.carlosnicolaugalves.makelifebetter.model.EventSection

sealed class EventResult {
    data object Idle : EventResult()
    data object Loading : EventResult()
    data class Success(val events: List<Event>) : EventResult()
    data class Error(val message: String) : EventResult()
}

sealed class EventSectionsResult {
    data object Idle : EventSectionsResult()
    data object Loading : EventSectionsResult()
    data class Success(val sections: List<EventSection>) : EventSectionsResult()
    data class Error(val message: String) : EventSectionsResult()
}

File: MakeLifeBetter/iosApp/iosApp/Features/Events/Domain/EventResult.swift
//
//  EventResult.swift
//  iosApp
//
//  Created by Carlos Nicolau Galves on 22/01/26.
//


enum EventResult {
    case idle
    case loading
    case success(events: [Event])
    case error(message: String)
}

enum EventSectionsResult {
    case idle
    case loading
    case success(sections: [EventSection])
    case error(message: String)
}